### PR TITLE
Migrate uyuni server to container

### PR DESCRIPTION
**Pull request by mbussolotto**
## What does this PR change?

Migrate a uyuni server to container. 
`uyuni-server-systemd-service` RPM already provide the script `uyuni-server.sh`. There's a new option called `migrate` to migrate a local uyuni server instance to container.

- [ ] UI works
- [ ] sync repository
- [ ] run highstate

Issue: https://github.com/SUSE/spacewalk/issues/20055

## GUI diff

No difference.

- [ ] **DONE**

## Documentation

- Documentation created https://github.com/uyuni-project/uyuni-docs/pull/2190

- [ ] **DONE**

## Test coverage
- No tests

- [x] **DONE**

## Links

- [ ] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: containers/server-image/Dockerfile [hunk 1]
```diff
@@ -83,3 +88,34 @@ LABEL org.opensuse.reference="${REFERENCE_PREFIX}/server:%PKG_VERSION%.%RELEASE%
...
+    -v tomcat:/etc/tomcat \
+    -v etc-tls:/etc/pki/tls \
+    -v ca-cert:/etc/pki/trust/anchors/ \
+    -v tls-key:/etc/pki/spacewalk-tls \
+    -v uyuni-config:/root \
+    -v tomcat-monitoring:/usr/lib/systemd/system/tomcat.service.d \
+    -v taskomatic-monitoring:/usr/lib/systemd/system/taskomatic.service.d \
+    -v etc-cobbler:/etc/cobbler \
+    -v var-lib-cobbler:/var/lib/cobbler \
+    -v home:/home \
```
Comment (aaannz) [inline containers/server-image/Dockerfile:119]: Why is home needed to be preserved?

File: containers/server-image/Dockerfile [hunk 2]
```diff
@@ -83,3 +88,34 @@ LABEL org.opensuse.reference="${REFERENCE_PREFIX}/server:%PKG_VERSION%.%RELEASE%
 
 CMD ["/usr/lib/systemd/systemd"]
 HEALTHCHECK --interval=5s --timeout=5s --retries=5 CMD ["/usr/bin/systemctl", "is-active", "multi-user.target"]
+
+LABEL MIGRATE="/usr/bin/podman run -it --rm     -v cgroup:/sys/fs/cgroup:rw \
```
Comment (aaannz) [inline containers/server-image/Dockerfile:92]: I don't remember, why we need cgroups volume?

File: rel-eng/build-packages-for-obs.sh [hunk 3]
```diff
@@ -88,9 +88,22 @@ while read PKG_NAME PKG_VER PKG_DIR; do
...
+      FAILED_CNT=$(($FAILED_CNT+1))
+      FAILED_PKG="$FAILED_PKG$(echo -ne "\n    $(basename $PKG_DIR)")"
+      echo "*** FAILED Building package [$(basename $PKG_DIR)] - $PKG_DIR does not exist"
+      continue 2
+    elif [ $CONTAINS_SPEC_FILE == 0 ]; then
+      #check if folder does not contains a spec file, otherwise although it's in containers folder it's a RPM package
+      CONTAINER_NAME=$(basename "$PKG_DIR")
+      echo "=== Building container image [${CONTAINER_NAME}]"
+
+      cp -r "$GIT_DIR/$PKG_DIR" "$SRPM_DIR/"
```
Comment (aaannz) [inline rel-eng/build-packages-for-obs.sh:106]: Why this change?

File: containers/server-image/Dockerfile (hunk 2 above)
Comment (mbussolotto) [reply containers/server-image/Dockerfile:92]: Actually I can't remember (not sure if @cbosdo can), but it's present also when we run container from scratch. I can test if it's really required.

File: rel-eng/build-packages-for-obs.sh (hunk 3 above)
Comment (mbussolotto) [reply rel-eng/build-packages-for-obs.sh:106]: These changes are already in Uyuni https://github.com/uyuni-project/uyuni/pull/6990 . Basically this commit https://github.com/uyuni-project/uyuni/commit/65901935cdcd5c3dd426950e46cd3af45f56f191 gaves the possibility to build container using  `build-packages-for-obs.sh` (instead of `build-containers-for-obs.sh` ) but it didn't work running with no argument. The PR fix the issue.

File: containers/server-image/Dockerfile (hunk 2 above)
Comment (cbosdo) [reply containers/server-image/Dockerfile]: IIRC this is needed to run systemd inside the container. Not sure if this is useful in the label.
